src/cls/rgw/cls_rgw.cc

```diff
@@ -949,6 +949,25 @@ int rgw_bucket_complete_op(cls_method_context_t hctx, bufferlist *in, bufferlist
       int ret = cls_cxx_map_set_val(hctx, idx, &new_key_bl);
       if (ret < 0)
 	return ret;
+
+      string instance_list_idx;
+      struct rgw_bucket_dir_entry instance_list_entry;
+      get_list_index_key(entry, &instance_list_idx);
+      int ret_tmp = read_index_entry(hctx, instance_list_idx, &instance_list_entry);
+      if (ret_tmp == -ENOENT) {
+      } else if (ret_tmp < 0) {
+      } else {
+        int flags_tmp = instance_list_entry.flags;
+        instance_list_entry = entry;
+        instance_list_entry.flags = flags_tmp;
```

Comment:
maybe instance_list_entry.meta.mtime = meta.mtime is more clear

Reply:
> maybe instance_list_entry.meta.mtime = meta.mtime is more clear

Yes, I do it at the beginning, It has almost no risk. But I think the values of [BI_BUCKET_OBJS_INDEX] and [BI_BUCKET_OBJ_INSTANCE_INDEX] should be the same(According to cls_rgw.cc:rgw_bucket_link_olh ---> cls_rgw.cc: write_obj_entries), then it caused another problem(due to "entry.flags = (entry.key.instance.empty() ? 0 : RGW_BUCKET_DIRENT_FLAG_VER);", current version is not visible, I am not sure what this line of code means)

[PS: In fact, at the beginning I want to call set_olh in the set_attrs function, Just like write_meta]